hpedockerplugin/volume_manager.py

```diff
@@ -398,22 +407,30 @@ def create_snapshot(self, src_vol_name, snapshot_name,
             response = json.dumps({'Err': msg})
             return response
 
-        return self._create_snapshot(src_vol_name, snapshot_name,
-                                     expiration_hrs, retention_hrs,
-                                     mount_conflict_delay)
+        return self._create_snapshot(src_vol_name, schedName, snapshot_name,
+                                     snapPrefix, expiration_hrs, exphrs,
+                                     retention_hrs, rethrs,
+                                     mount_conflict_delay, has_schedule,
+                                     schedFrequency, current_backend)
 
     @synchronization.synchronized('{src_vol_name}')
-    def _create_snapshot(self, src_vol_name, snapshot_name,
-                         expiration_hrs, retention_hrs,
-                         mount_conflict_delay):
+    def _create_snapshot(self, src_vol_name, schedName, snapshot_name,
+                         snapPrefix, expiration_hrs, exphrs, retention_hrs,
+                         rethrs, mount_conflict_delay, has_schedule,
+                         schedFrequency, current_backend):
+
         vol = self._etcd.get_vol_byname(src_vol_name)
         if vol is None:
             msg = 'source volume: %s does not exist' % src_vol_name
             LOG.debug(msg)
             response = json.dumps({u"Err": msg})
             return response
-
         volid = vol['id']
+        if 'has_schedule' not in vol:
+            vol_sched_flag = volume.DEFAULT_SCHEDULE
+            vol['has_schedule'] = vol_sched_flag
+            self._etcd.update_vol(volid, 'has_schedule', vol_sched_flag)
```

Comment:
Along similar lines, might want to populate default backend name for the legacy volume(s) meta-data. I understand that the code is capable of handling legacy volumes - so this is just a suggestion from consistency perspective of how we have been handling updates to legacy volumes to have new information.

Reply:
Will do this.